### DISCUSSION TITLE
[Feature] Hide album share suggestions

### DISCUSSION THREAD

**Discussion by myadoran** [immich-app/immich#4408] [Feature Request]
### The feature

Our instance is used by a diverse group of people with little connection to each other. Upon using the "Invite to album" feature a list of other users on the instance is shown. I think it would be valuable to have the option of hiding this suggestion list, preventing users from finding each other's name and email address.

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by cookieflips** [comment]
Seconding this. I'm using my server to host two groups of people, would be great if all the users aren't automatically revealed to each other!

**Comment by myadoran** [comment]
@cookieflips heya! I've been using a workaround for a while that seems to work well. If you've got a reverse proxy in front of your instance, you can block the /api/users endpoints while allowing self-identification (otherwise things surrounding logging in **do** break :) ).

For my nginx setup that looks something like this:

```
        ##DENY USER LISTING
        location /api/users {
                deny all;
                proxy_pass http://host:port/api/users;
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        }
        ##ALLOW USER SELF IDENTIFICATION
        location /api/users/me {
                allow all;
                proxy_pass http://host:port/api/users/me;
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        }
```

This results in users still being able to share through links, but just not being able to see the other people.

![image](https://github.com/user-attachments/assets/60083af8-c4b9-486d-a9eb-3a82316e1607)

If you'd like to try this but need some pointers, just let me know!


Mya

**Comment by MK73DS** [comment]
I would also love such a feature. Ideally, I would like the administrator to be able to set 3 privileges for a user : 
- Can add any other Immich user as a partner (selection from a list of all users)
- Can add any other Immich user as a partner (by entering the email address of another user, doesn't have access to a list of all users)
- Cannot add any other Immich user as a partner.

**Comment by bo0tzz** [comment]
This is now possible by turning off the public users setting in https://my.immich.app/admin/system-settings?isOpen=server.
